include/atomics/sender_cadmium.hpp

```diff
@@ -0,0 +1,210 @@
+/**
+ *Cristina Ruiz Martin
+ *ARSLab - Carleton University
+ */
+
+#ifndef BOOST_SIMULATION_PDEVS_SENDER_HPP
+
+#define BOOST_SIMULATION_PDEVS_SENDER_HPP
+#include <cadmium/modeling/ports.hpp>
+#include <cadmium/modeling/message_bag.hpp>
+#include <limits>
+#include <math.h>
+#include <assert.h>
+#include <memory>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <algorithm>
+#include <limits>
+#include <random>
+#include "../data_structures/message.hpp"
+using namespace cadmium;
+using namespace std;
+
+//Port definition
+struct sender_defs
+{
+    struct packetSentOut: public out_port <message_t> {};
+    struct ackReceivedOut: public out_port <message_t> {};
+    struct dataOut: public out_port <message_t> {};
+    struct controlIn: public in_port <message_t> {};
+    struct ackIn: public in_port <message_t> {};
+};
+
+template <typename TIME>
+    class Sender
+    {
+        using definitions = sender_defs; // putting definitions in context
+        public:
+            //Parameters to be overwriten when instantiating the atomic model
+            TIME PREPARATION_TIME;
+        TIME timeout;
+        // default constructor
+        Sender() noexcept
+        {
+            PREPARATION_TIME = TIME("00:00:10");
+            timeout = TIME("00:00:20");
+            state.alt_bit = 0;
+            state.next_internal = std::numeric_limits <TIME>::infinity();
+            state.model_active = false;
+        }
+
+        // state definition
+        struct state_type
+        {
+            bool ack;
+            int packet_number;
+            int total_packet_number;
+            int alt_bit;
+            bool sending;
+            bool model_active;
+            TIME next_internal;
+        };
+        state_type state;
+        // ports definition
+        using input_ports = std::tuple <typename definitions::controlIn, 
+																typename definitions::ackIn> ;
+        using output_ports = std::tuple <typename definitions::packetSentOut, 
+																typename definitions::ackReceivedOut, typename definitions::dataOut> ;
+
+        // internal transition
+        void internal_transition()
+        {
+            if (state.ack)
+            {
+                if (state.packet_number < state.total_packet_number)
+                {
```

Comment:
All if statement  needs to follow coding conventions as specified by professor:
ie,
if (value == 1) {
   value = 2;
}
pls, apply to all if statement code convention  in your code